Mathlib/LinearAlgebra/Matrix/ToLin.lean

```diff
@@ -885,6 +885,12 @@ theorem toMatrix_distrib_mul_action_toLinearMap (x : R) :
     Pi.single_apply]
 #align to_matrix_distrib_mul_action_to_linear_map toMatrix_distrib_mul_action_toLinearMap
 
+lemma LinearMap.toMatrix_prodMap [DecidableEq n] [DecidableEq m] [DecidableEq (n ⊕ m)]
```

Comment:
This isn't necessary, right? We have [instDecidableEqSum](https://leanprover-community.github.io/mathlib4_docs/Init/Core.html#instDecidableEqSum)
```suggestion
lemma LinearMap.toMatrix_prodMap [DecidableEq n] [DecidableEq m]
```

Reply:
I think @urkud argued that we shouldn't rely on that instance. Yury, can you confirm?

Reply:
I think @jcommelin is right here: we should be unopinionated about data in lemmas.

For example if we have a definition consuming terms of type `Matrix ι ι R` whose statement needs `[DecidableEq ι]` and we want to combine this lemma with it in the case `ι = n ⊕ m` then we need this extra flexibility.